sql/schema.go

```diff
@@ -0,0 +1,215 @@
+package sql
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"go.uber.org/zap"
+)
+
+const (
+	SchemaPath        = "schema/schema.sql"
+	UpdatedSchemaPath = "schema/schema.sql.updated"
+)
+
+// LoadDBSchemaScript retrieves the database schema as text.
+func LoadDBSchemaScript(db Executor) (string, error) {
+	var (
+		err error
+		sb  strings.Builder
+	)
+	version, err := version(db)
+	if err != nil {
+		return "", err
+	}
+	fmt.Fprintf(&sb, "PRAGMA user_version = %d;\n", version)
+	if _, err = db.Exec(
+		// Type is either 'index' or 'table', we want tables to go first
+		`select tbl_name, sql || ';' from sqlite_master
+                 where sql is not null
+                 order by tbl_name, type desc, name`,
+		nil, func(st *Statement) bool {
+			fmt.Fprintln(&sb, st.ColumnText(1))
+			return true
+		}); err != nil {
+		return "", fmt.Errorf("error retrieving DB schema: %w", err)
+	}
+	// On Windows, the result contains extra carriage returns
+	return strings.ReplaceAll(sb.String(), "\r", ""), nil
+}
+
+// Schema represents database schema.
+type Schema struct {
+	Script        string
+	Migrations    MigrationList
+	skipMigration map[int]struct{}
+}
+
+// Diff diffs the database schema against the actual schema.
+// If there's no differences, it returns an empty string.
+func (s *Schema) Diff(actualScript string) string {
+	return cmp.Diff(s.Script, actualScript)
+}
```

Comment:
Would it make sense to ignore whitespace?

```suggestion
// Diff diffs the database schema against the actual schema.
// If there's no differences, it returns an empty string.
func (s *Schema) Diff(actualScript string) string {
	// ignore whitespace differences
	opt := cmp.Comparer(func(x, y string) bool {
		return strings.Join(strings.Fields(x), "") == strings.Join(strings.Fields(y), "")
	})
	return cmp.Diff(s.Script, actualScript, opt)
}
```

Reply:
Fixed, thanks